cli/cmd/root.go

```diff
@@ -189,6 +191,30 @@ func getSuccessRate(success, failure uint64) float64 {
 	return float64(success) / float64(success+failure)
 }
 
+// getDefaultNamespace fetches the default namespace
+// used in the current KubeConfig context
+func getDefaultNamespace() string {
+	rules := clientcmd.NewDefaultClientConfigLoadingRules()
+
+	if kubeconfigPath != "" {
+		rules.ExplicitPath = kubeconfigPath
+	}
+
+	overrides := &clientcmd.ConfigOverrides{CurrentContext: kubeContext}
+	kubeCfg := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, overrides)
+	ns, _, err := kubeCfg.Namespace()
+
+	if err != nil {
+		log.Errorf("failed to set namespace from config context:%v", err)
+	}
```

Comment:
Cant you just return the defaultContextNamespace and call it a day instead of doing another check for it being empty further down?

Reply:
Can do, yes. This is how it was done initially, so I will add it back in. Alternatively, another solution that I've been hesitant on would be to just log the error and return the `ns`. If there is an error, the `ns` would be `""` which is equal to the default anyway. I thought initially this approach broke the unit tests but it doesn't. Let me know if that sounds better and I'll put it in.